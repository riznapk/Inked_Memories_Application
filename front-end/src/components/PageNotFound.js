import React from "react";
import { Typography } from "@mui/material";
import { Link } from "react-router-dom";
import styled from "styled-components";

const PageNotFoundContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
  background-color: #252525;
`;

const Title = styled.h1`
  font-size: 3rem;
  font-weight: bold;
  color: #797d62;
  margin-bottom: 1rem;
`;

const Message = styled.p`
  font-size: 1.5rem;
  color: #fff;
`;

function PageNotFound() {
  return (
    <PageNotFoundContainer>
      <Title>404</Title>
      <Message>Page not found</Message>
      <Typography sx={{ color: "white" }}>
        Go To Home Page {""}
        <Link to={"/"} style={{ color: "#797d62" }}>
          Click here
        </Link>
      </Typography>
    </PageNotFoundContainer>
  );
}

export default PageNotFound;
