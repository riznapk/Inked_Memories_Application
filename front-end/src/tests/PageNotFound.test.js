import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

import PageNotFound from "../components/PageNotFound";

test("PageNotFound component testing", () => {
  const component = renderer.create(
    <MemoryRouter>
      <PageNotFound />
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
