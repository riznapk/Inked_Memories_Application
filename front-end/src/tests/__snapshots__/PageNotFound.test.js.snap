// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound component testing 1`] = `
<div
  className="sc-aXZVg cuTzES"
>
  <h1
    className="sc-gEvEer epkMoJ"
  >
    404
  </h1>
  <p
    className="sc-eqUAAy erewlt"
  >
    Page not found
  </p>
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-jb1ygz-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;color:white;}",
      }
    }
    data-emotion="css jb1ygz-MuiTypography-root"
  />
  <p
    className="MuiTypography-root MuiTypography-body1 css-jb1ygz-MuiTypography-root"
  >
    Go To Home Page 
    <a
      href="/"
      onClick={[Function]}
      style={
        {
          "color": "#797d62",
        }
      }
    >
      Click here
    </a>
  </p>
</div>
`;
